### PR TITLE
ARROW-2911: [Python] Parquet binary statistics that end in '\0' truncate last byte

### PR DESCRIPTION
**Pull request by xhochy**
Depends on https://github.com/apache/parquet-cpp/pull/479

Merge once parquet-cpp 1.5.0 is released.

### CODE REVIEW

**Comment by wesm** [comment]
Thanks. I want to bring up the Arrow-Parquet monorepo discussion again. How much better would our lives be if we were able to develop this software all in the same place without these release and API change headaches? I keep thinking about ways to get rid of the build system duplication also but I don't have great ideas

**Comment by xhochy** [comment]
I have also thought a lot about this as I have some fears that this repository grows too large but currently I think I'm confident in getting the Arrow build faster than improving the build structure with the two repositories. 

Still, we should not convert this on the fast path to a monorepo. This would be a hard-to-reverse decision. I hope to have some time in late August to write a proposal how we could integrate both into a repo and what we need to do to keep this viable. My greatest fear is that the build will be even more red than it is currently.

**Comment by wesm** [comment]
Almost all contributors (> 5 patches) who contribute to parquet-cpp have also contributed to Arrow. The only parquet-cpp committer who is not an Arrow committer is @majetideepak, but that could be easily fixed.

I think we need to be laying plans to accommodate a C++ codebase in Arrow that exceeds 1 million lines (we are a bit under 100,000 right now). This may require a mix of lighter-weight "every commit" CI in Travis CI combined with more exhaustive unit testing in Jenkins. As two projects I've seen, Kudu and Impala have significantly longer build times and much more complex and time consuming test procedures to validate commits. Given the other challenges we've dealt with so far, scaling our build and test process seems well within our reach.

What I don't really know is how to best handle the "Apache Arrow" vs "Apache Parquet" entity distinctions. There is enough overlap between committers and PMC members that we could work something out that is in the interest of the community's productivity and health.

In summary, building open source software is difficult enough already. If we can ease our burden and make ourselves more productive, I think we should try to do that.

**Comment by pitrou** [comment]
This has conflicts and needs rebasing.

**Comment by xhochy** [comment]
cc @cpcloud this should fix the master failure.

File: python/pyarrow/_parquet.pxd [hunk 1]
```diff
@@ -135,6 +135,7 @@ cdef extern from "parquet/api/schema.h" namespace "parquet" nogil:
         int num_columns()
 
     cdef c_string FormatStatValue(ParquetType parquet_type, const char* val)
```
Comment (pitrou) [inline python/pyarrow/_parquet.pxd]: How about removing the deprecated version?

**Review by pitrou — APPROVED** [review]
LGTM, just a comment.

File: python/pyarrow/_parquet.pxd (hunk 1 above)
Comment (xhochy) [reply python/pyarrow/_parquet.pxd]: will do

**Comment by xhochy** [comment]
AppVeyor: https://ci.appveyor.com/project/xhochy/arrow/build/1.0.723
